stackslib/src/burnchains/bitcoin/blocks.rs

```diff
@@ -150,9 +150,7 @@ impl BitcoinMessageHandler for BitcoinBlockDownloader {
             None => panic!("No block header set"),
             Some(ref ipc_header) => {
                 let block_hash = ipc_header.block_header.header.bitcoin_hash().clone();
-                indexer
-                    .send_getdata(&vec![block_hash])
-                    .and_then(|_r| Ok(true))
+                indexer.send_getdata(&[block_hash]).and_then(|_r| Ok(true))
```

Comment:
No need to use `and_then()` if you always return `Ok()`

```suggestion
                indexer.send_getdata(&[block_hash]).map(|_| true)
```

Reply:
This gets fixed by another Clippy warning. Leaving for now.